gcloud-java-compute/src/main/java/com/google/gcloud/compute/SnapshotDiskConfiguration.java

```diff
@@ -0,0 +1,165 @@
+
+  private static final long serialVersionUID = -1996055058706221049L;
+
+  private final SnapshotId sourceSnapshot;
+  private final String sourceSnapshotId;
+
+  /**
+   * A builder for {@code SnapshotDiskConfiguration} objects.
+   */
+  public static class Builder
+      extends DiskConfiguration.Builder<SnapshotDiskConfiguration, Builder> {
+
+    private SnapshotId sourceSnapshot;
+    private String sourceSnapshotId;
+
+    private Builder() {
+      super(Type.SNAPSHOT);
+    }
+
+    private Builder(SnapshotDiskConfiguration snapshotDiskInfo) {
+      super(Type.SNAPSHOT, snapshotDiskInfo);
+      this.sourceSnapshot = snapshotDiskInfo.sourceSnapshot;
+      this.sourceSnapshotId = snapshotDiskInfo.sourceSnapshotId;
+    }
+
+    private Builder(Disk diskPb) {
+      super(Type.SNAPSHOT, diskPb);
+      this.sourceSnapshot = SnapshotId.fromUrl(diskPb.getSourceSnapshot());
+      this.sourceSnapshotId = diskPb.getSourceSnapshotId();
+    }
+
+    /**
+     * Sets the identity of the source snapshot used to create the disk.
+     */
+    public Builder sourceSnapshot(SnapshotId sourceSnapshot) {
+      this.sourceSnapshot = checkNotNull(sourceSnapshot);
+      return this;
+    }
+
+    Builder sourceSnapshotId(String sourceSnapshotId) {
+      this.sourceSnapshotId = sourceSnapshotId;
+      return this;
+    }
+
+    /**
+     * Creates a {@code SnapshotDiskConfiguration} object.
+     */
+    @Override
+    public SnapshotDiskConfiguration build() {
+      return new SnapshotDiskConfiguration(this);
+    }
+  }
+
+  private SnapshotDiskConfiguration(Builder builder) {
+    super(builder);
+    this.sourceSnapshot = checkNotNull(builder.sourceSnapshot);
+    this.sourceSnapshotId = builder.sourceSnapshotId;
+  }
+
+  /**
+   * Returns the identity of the source snapshot used to create the disk.
+   */
+  public SnapshotId sourceSnapshot() {
+    return sourceSnapshot;
+  }
+
+  /**
+   * Returns the unique ID of the snapshot used to create this disk. This value identifies the exact
+   * snapshot that was used to create the persistent disk. For example, if you created the
+   * persistent disk from a snapshot that was later deleted and recreated under the same name, the
+   * source snapshot ID would identify the exact version of the snapshot that was used.
+   */
+  public String sourceSnapshotId() {
+    return sourceSnapshotId;
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  MoreObjects.ToStringHelper toStringHelper() {
+    return super.toStringHelper()
+        .add("sourceSnapshot", sourceSnapshot)
+        .add("sourceSnapshotId", sourceSnapshotId);
+  }
+
+  @Override
+  public final int hashCode() {
+    return Objects.hash(baseHashCode(), sourceSnapshot, sourceSnapshotId);
+  }
+
+  @Override
+  public final boolean equals(Object obj) {
+    return obj instanceof SnapshotDiskConfiguration && baseEquals((SnapshotDiskConfiguration) obj);
```

Comment:
Since `baseEquals` checks if the runtime classes are the same, does that mean that the first half of this boolean statement is unnecessary?